### PR TITLE
added genesis config option ecip1017EraRounds for Ethereum Classic chains

### PR DESCRIPTION
**Pull request by edwardmack**
## PR description
Ethereum Classic [ECIP-1017](https://ecips.ethereumclassic.org/ECIPs/ecip-1017) Monetary Policy and Final Modification to the Ethereum Classic Emission Schedule introduces the concept of an "Era" which is defined as the number of blocks containing a given production rate.  The Ethereum Classic Network uses a value of 5,000,000 blocks for an Era.

This PR introduces a genesis config option named `ecip1017EraRounds` to define the number of Rounds in an Era.  This config value is used in the mordor.json config since the Mordor testnet uses a 2,000,000 block Era.

If `ecip1017EraRounds` is not configured, the node will default to 5,000,000 blocks per Era.

Signed-off-by: Edward Mack <ed@edwardmack.com>

## Fixed Issue(s)

Fixes issue where node would not sync past 2,000,000 blocks on Mordor testnet.

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: config/src/main/java/org/hyperledger/besu/config/GenesisConfigOptions.java [hunk 1]
```diff
@@ -160,6 +160,18 @@
 
   OptionalInt getEvmStackSize();
 
+  /**
+   * Number of rounds contained within an Era for calculating Ethereum Classic Emission Schedule,
+   * ECIP defines this as 5,000,000 however this config option allows for adjusting (for using with
+   * other networks, for example Mordor testnet uses 2,000,000). The values defaults to 5,000,000 in
```
Comment (shemnon) [inline config/src/main/java/org/hyperledger/besu/config/GenesisConfigOptions.java:166]:
```suggestion
   * other networks, for example Mordor testnet uses 2,000,000). The values defaults to 5,000,000 if
```

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ClassicBlockProcessor.java [hunk 2]
```diff
@@ -31,21 +32,25 @@
 
   private static final Logger LOG = LogManager.getLogger();
 
-  private static final long ERA_LENGTH = 5_000_000L;
+  private long eraLength = 5_000_000L;
```
Comment (shemnon) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ClassicBlockProcessor.java:35]:
```suggestion
  private static final long DEFAULT_ERA_LENGTH = 5_000_000L;

  private final long eraLength;
```

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ClassicBlockProcessor.java [hunk 3]
```diff
@@ -31,21 +32,25 @@
...
     super(
         transactionProcessor,
         transactionReceiptFactory,
         blockReward,
         miningBeneficiaryCalculator,
         skipZeroBlockRewards,
         TransactionGasBudgetCalculator.frontier());
+    if (eraLen != null && eraLen.isPresent()) {
+      eraLength = eraLen.getAsLong();
+    }
```
Comment (shemnon) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ClassicBlockProcessor.java:53]:
``` suggestion
    eraLength = eraLen.orElse(DEFAULT_ERA_LENGTH);
```

**Comment by bobsummerwill** [comment]
Thanks for squeezing this in, guys.

Just spotted this typo while looking at the diffs - DEFALUT_ERA_LENGTH

Best wishes!

**Comment by edwardmack** [comment]
@bobsummerwill  thanks for catching the typo, fixed in PR #1343

**Pull request merged**